src/item_factory.cpp

```diff
@@ -710,24 +710,15 @@ void Item_factory::load( islot_gun &slot, JsonObject &jo )
         }
     }
 
-    //Add any built-in mods.
-    if( jo.has_array( "built_in_mods" ) ) {
-    JsonArray jarr = jo.get_array( "built_in_mods" );
-        while( jarr.has_more() ) {
-            std::string temp = jarr.next_string();
-            slot.built_in_mods.push_back( temp );
-        }
+    JsonArray builtmod = jo.get_array( "built_in_mods" );
+    while( builtmod.has_more() ) {
+        slot.built_in_mods.emplace( builtmod.next_string() );
```

Comment:
I made the change and am building.
Avoiding copy construction isn't really compelling unless it happens in a
tight loop, so I'd default to insert. Also it might keep a move from
happening sometimes.